### PR TITLE
fix(spec): make the agent-tag pre-flight match what the tooling emits

### LINKED ISSUE #769: BUG-041: dotf spec archive pre-flight false-positives on [AGENT-DRAFT] inside code spans and completed tasks
Found during the #670 sweep (PR #768).

## Problem

`dotf spec archive`'s pre-flight refuses when a spec mentions `[AGENT-DRAFT]` or `[AGENT-SUGGESTION]` **as quoted content**, not as an unresolved marker:

```console
$ dotf spec archive HARNESS-026-session-brief-core
Error: unresolved [AGENT-DRAFT]/[AGENT-SUGGESTION] tags found:
  tasks.md:25: - [x] Add `sb_specs` (active/archived counts + `[AGENT-DRAFT]` flagging — lifted from `detect_repo_specs`)
resolve them (accept/edit/delete) before archiving, or use --force-with-drafts
```

The flagged line is a **completed task** (`- [x]`) whose text sits inside a **code span**, describing a feature that *reports* AGENT-DRAFT tags. There is nothing to resolve. The check is a plain substring match with no awareness of backticks or checkbox state.

The irony is structural: any spec for tooling that handles draft markers cannot be archived without `--force-with-drafts`, and this repo builds exactly that tooling.

## Impact

Low severity, but it pushes users toward `--force-with-drafts` as a habit, which defeats the guard: once the flag is reflexive, a *genuine* unresolved draft sails through too. A guard that cries wolf is training people to ignore it.

## Proposed fix

Narrow the match so it cannot fire on quoted or completed content:

- Ignore occurrences inside backtick code spans and fenced code blocks.
- Ignore lines that are already-ticked checklist items (`- [x]`), which by definition are resolved.
- Consider requiring the marker at a line-leading position, matching how the tags are actually emitted.

Whichever combination is chosen, the test must **plant a real unresolved tag and confirm the pre-flight still refuses** — the guard has to be verified in the red direction, not only the green one (`docs/lessons.md`, 2026-08-05).

## Acceptance criteria

- A spec containing `` `[AGENT-DRAFT]` `` inside a code span, or in a `- [x]` line, archives without `--force-with-drafts`.
- A spec with a genuine unresolved `[AGENT-DRAFT]` marker still refuses.
- Both directions covered by a Go test in `cli/internal/spec`.

## Context

Reproduced on `specs/HARNESS-026-session-brief-core/tasks.md:25`. That spec was archived in #768 with `--force-with-drafts` after confirming by eye that the tag was documentation, not a draft.

A second, unrelated observation from the same sweep, worth recording where the fix will be made: **`dotf spec archive` exits 1 on this refusal but prints to stderr in a shape that is easy to mistake for success when scripted.** A batch loop matching output text rather than the exit code will report a failed archive as OK. Prefer `$?` over output matching in any wrapper — and the tool could help by making the failure line unambiguous.

### PR DESCRIPTION
**Pull request by mlorentedev**
## Problem

The archive lock was inverted in **both** directions, and the reported symptom was only the milder half.

**Reported (false positive):** the pre-flight refuses to archive a spec that merely *quotes* the editorial markers — inside a code span, or on a completed `- [x]` line. That trains people to reach for `--force-with-drafts`, and once the flag is reflexive a genuine draft sails through too.

**Found while fixing it, and worse (false negative):** the pattern required `]` immediately after the keyword. `/spec fill` emits the *suffixed* form documented in `harness/skills/spec/SKILL.md`, which that pattern has never matched — so the archive lock has never locked on a marker emitted the way the tooling emits it. Proven against the tree, not by reasoning:

```console
$ grep -o '\[AGENT-SUGGESTION[^]]*\]' specs/archive/CLI-002-repo-structure/proposal.md
[AGENT-SUGGESTION — accept or remove]
```

That spec is **archived**, carrying a live marker the lock never saw.

Stated plainly: **the guard matched exactly the shape that is always a false positive, and was blind to exactly the shape that is always a true positive.**

A second call site had drifted independently — the session-start injector used a cruder literal substring scan, so it and the archiver disagreed about what "unresolved" means. It flagged `AI-028-hive-install-model-migration` on the strength of a *completed* task whose own text says the marker is resolved.

## Fix

Widen the pattern to both emitted shapes; exclude occurrences that are quoted rather than live (fenced blocks, inline code spans, ticked `- [x]` lines); and give both call sites one predicate, `spec.ScanUnresolvedTags`, so they cannot disagree again.

Shape-based exclusion rather than a markdown parse, deliberately: a spec is not arbitrary markdown, and a parser would be a far heavier dependency than the two shapes that actually produce false positives here.

## This PR is blocked by its own bug

The unpatched binary refuses to archive this very spec — on the line that quotes the issue title:

```console
$ dotf spec archive BUG-041-spec-tag-preflight       # installed (unpatched) build
Error: unresolved [AGENT-DRAFT]/[AGENT-SUGGESTION] tags found:
  proposal.md:15: <!-- from issue #769: BUG-041: dotf spec archive pre-flight
  false-positives on [AGENT-DRAFT] inside code spans and completed tasks -->
```

The bug blocking its own fix, on a line whose content is the bug's description.

Note what the fix does **not** do: it does not wave that line through. A bare marker in live prose is a real marker, and an HTML comment is prose. The resolution is the one the tooling intends — quote it as a code span, because it is a quotation of an issue title, not an editorial directive. The patched binary then archives cleanly. That is the workflow this change restores.

## Live repro, fixed

Session-start used to emit `1 with unresolved [AGENT-DRAFT]/[AGENT-SUGGESTION] tags: - AI-028-hive-install-model-migration`. The patched binary emits `[specs] 29 active, 119 archived` with no clause at all.

## Coverage

`TestScanUnresolvedTags`, table-driven over both directions — four "must not fire" shapes (code span, ticked item, fenced block, tilde-fenced block) and six "must fire" (both canonical forms, the bare form in a comment, an unticked item, after a closed fence, and with a longer closing fence). Plus two integration cases at the level the user experiences, including the red direction: a spec carrying the canonical emitted form must still be refused **by name**, or a scanner matching nothing at all would satisfy every green case.

`gofmt` clean on the touched files; `go build ./...` and `go vet ./...` pass; full Go suite green across 12 packages.

## Note on sequencing

This PR is over the LOC threshold and closes its own issue, so it was **unlandable** until #808 merged — archiving satisfied one half of spec-gate and broke the other. It is now the first real consumer of #808's fix, and specifically of its "a spec created and archived in the same PR counts" case.

Closes #769

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved draft-tag detection during session startup and specification archiving.
  * Recognizes bare and suffixed draft or suggestion markers.
  * Ignores markers in code blocks, inline code, and completed checklist items.
  * Preserves accurate source lines and paths when reporting unresolved tags.

* **Tests**
  * Added coverage for tag detection, archive validation, and Markdown edge cases.

* **Documentation**
  * Added archived specifications and verification notes for the tag preflight fixes.

### CODE REVIEW

File: cli/internal/spec/archive.go [hunk 1]
```diff
@@ -10,9 +10,67 @@ import (
...
+// opening one (CommonMark), and of the same character.
+var fenceLinePattern = regexp.MustCompile("^\\s{0,3}(`{3,}|~{3,})")
+
+// completedTaskPattern matches a ticked checklist item. A tag on such a line is
+// by definition resolved — the line records work already done.
+var completedTaskPattern = regexp.MustCompile(`^\s*[-*+]\s+\[[xX]\]`)
+
+// codeSpanPattern matches an inline code span, so its contents can be dropped
+// before scanning. Backtick runs are matched symmetrically enough for prose.
+var codeSpanPattern = regexp.MustCompile("`+[^`]*`+")
```
[the remark this answers is not in the document]
Comment (mlorentedev) [reply cli/internal/spec/archive.go]: Confirmed and fixed in `445f242`. Verified by execution before accepting it:

```text
"a `[AGENT-DRAFT]` b"      -> "a  b"      (valid span, correctly stripped)
"a `[AGENT-DRAFT]`` b"     -> "a  b"      (NOT a span -- marker silently hidden)
"a ``[AGENT-DRAFT]` b"     -> "a  b"      (NOT a span -- marker silently hidden)
```

Rows two and three are not code spans, so their markers are live -- this was the same false-negative class the PR exists to remove, reintroduced by the PR's own exclusion rule.

RE2 has no backreferences, so the balanced-run rule cannot be expressed as a regexp at all. Replaced with `stripCodeSpans`, which scans for a closing run of exactly the opening length. On the multi-line point: a run with no equal-length partner is now left in place, so an unbalanced or line-crossing backtick makes the scanner **report** rather than skip -- deliberately erring toward refusing an archive, never toward passing one. Documented as such rather than left implicit.

Three cases pin it: mismatched runs still fire, an unterminated run leaves the line scannable, and a balanced double-backtick span is still excluded.

File: specs/archive/BUG-041-spec-tag-preflight/tasks.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+- [x] `proposal.md` complete, acceptance criteria testable
+
+## Implementation
+
+- [x] Establish the canonical emitted form from `harness/skills/spec/SKILL.md`
+      rather than from the existing pattern, which is the artefact under suspicion
+- [x] Confirm the false negative against the tree, not by reasoning: grep the
+      archive for the suffixed form and find `CLI-002-repo-structure` carrying one
+- [x] [AC2][AC3] Widen the pattern to both shapes
+- [x] [AC1][AC4] `ScanUnresolvedTags` — fence tracking, code-span stripping,
```
[the remark this answers is not in the document]
Comment (mlorentedev) [reply specs/archive/BUG-041-spec-tag-preflight/tasks.md:23]: Not changing this one, deliberately. markdownlint is not in this repo's CI (no workflow or pre-commit hook references it), and the `[AC1]`-style acceptance-criterion labels are the established convention across the spec tree -- `specs/BUG-049-test-sh-repo-root/tasks.md` carries 8 of them and the already-merged `specs/archive/BUG-050-spec-gate-satisfiable/tasks.md` carries 9. Escaping them here alone would make this spec the inconsistent one.

If the convention is worth changing, it is worth changing tree-wide in its own PR rather than drifting one spec at a time -- happy to file that if you think the reference-link ambiguity is worth the churn.

File: specs/archive/BUG-041-spec-tag-preflight/verification.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+### The live repro for the reported symptom
+
+Session-start flagged `AI-028-hive-install-model-migration` as carrying an
+unresolved marker. The sole occurrence is at `tasks.md:17`, inside a **completed**
+checkbox whose text states the marker is resolved, and inside a code span.
+
+After the fix, the patched binary emits:
+
+```
```
[the remark this answers is not in the document]
Comment (mlorentedev) [reply specs/archive/BUG-041-spec-tag-preflight/verification.md]: Fixed in `445f242` -- the block is now fenced as `text`.
